assets/queries/terraform/aws/lambda_iam_invokefunction_misconfigured/query.rego

```diff
@@ -0,0 +1,47 @@
+package Cx
+
+import data.generic.common as common_lib
+import data.generic.terraform as tf_lib
+
+#CxPolicy for ressource iam policy
+CxPolicy[result] {
+	resourceType := {"aws_iam_role_policy", "aws_iam_user_policy", "aws_iam_group_policy", "aws_iam_policy"}
+	resource := input.document[i].resource[resourceType[idx]][name]
+	policy := common_lib.json_unmarshal(resource.policy)
+	st := common_lib.get_statement(policy)
+	statement := st[_]
+
+
+	check_iam_action(statement) == true
+	not check_iam_ressource(statement)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("aws_iam_policy[{{%s}}]", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("aws_iam_policy[%s].statement.ressource is misconfigured", [name]),
+		"keyActualValue": sprintf("aws_iam_policy[%s].statement.ressource allow access to function (unqualified ARN) and its sub-resources, add another statement with :* to function name", [name])
+	}
```

Comment:
Please, note that:
- `resourceType` and `resourceName` are required in the results of policies of TF queries.
- we should point the correct resource in results
- we should point the searchKey to the field more adequate. In this case, policy

```suggestion
    result := {
		"documentId": input.document[i].id,
		"resourceType": resourceType[idx],
		"resourceName": tf_lib.get_resource_name(resource, name),
		"searchKey": sprintf("%s[%s].policy", [resourceType[idx], name]),
		"issueType": "IncorrectValue",
		"keyExpectedValue": sprintf("%s[%s].policy is misconfigured", [name]),
		"keyActualValue": sprintf("%s[%s].policy allows access to function (unqualified ARN) and its sub-resources, add another statement with \":*\" to function name", [name])
	}
```